src/backend/GnuCashBudget.GnuCashData.Generator/Services/ScopedBackgroundService.cs

```diff
@@ -0,0 +1,37 @@
+using GnuCashBudget.GnuCashData.Generator.Interfaces;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+
+namespace GnuCashBudget.GnuCashData.Generator.Services;
+
+public sealed class ScopedBackgroundService(
+    IServiceScopeFactory serviceScopeFactory,
+    ILogger<ScopedBackgroundService> logger) : BackgroundService
+{
+    private const string ClassName = nameof(ScopedBackgroundService);
+
+    protected override async Task ExecuteAsync(CancellationToken cancellationToken)
+    {
+        logger.LogInformation("{Name} is running", ClassName);
+
+        await DoWorkAsync(cancellationToken);
+    }
+
+    private async Task DoWorkAsync(CancellationToken cancellationToken)
```

Comment:
Not sure about this usage of BackgroundService. When I run the app it works but it does not end automatically. Is this correct? @veselfil

Reply:
Removed IHostService and now it works as intended. Thanks @veselfil for the tip